tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,339 @@
+	require.NoError(c.T(), err)
+
+	scanName := namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+
+	// Run the scan
+	err = runScan(client, scanName, c.cisProfileName, c.project.ClusterID)
+	require.NoError(c.T(), err)
+
+}
+
+func (c *CisBenchmarkTestSuite) TestUpgradeCisbenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	// Change cisbenchmark install option version to previous version of the latest version
+	versionsList, err := client.Catalog.GetListChartVersions(charts.CISBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+
+	if len(versionsList) < 2 {
+		c.T().Skip("Skipping the upgrade case, only one version of cis-benchmark is available")
+	}
+	assert.GreaterOrEqualf(c.T(), len(versionsList), 2, "There should be at least 2 versions of the cis-benchmark chart")
+	versionLatest := versionsList[0]
+	c.T().Log(versionLatest)
+
+	versionBeforeLatest := versionsList[1]
+	c.T().Log(versionBeforeLatest)
+	c.chartInstallOptions.Version = versionBeforeLatest
+
+	c.T().Log("Checking if the cis-benchmark chart is installed with one of the previous versions")
+	initialCisbenchmarkChart, err := extencharts.GetChartStatus(client, c.project.ClusterID, charts.CISBenchmarkNamespace, charts.CISBenchmarkName)
+	require.NoError(c.T(), err)
+
+	if initialCisbenchmarkChart.IsAlreadyInstalled && initialCisbenchmarkChart.ChartDetails.Spec.Chart.Metadata.Version == versionLatest {
+		c.T().Skip("Skipping the upgrade case, cis-benchmark chart is already installed with the latest version")
+	}
+
+	if !initialCisbenchmarkChart.IsAlreadyInstalled {
+		c.T().Log("Installing cis-benchmark chart with the version before the latest version")
+		err = charts.InstallCISBenchmarkChart(client, c.chartInstallOptions)
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cis-benchmark chart deployments to have expected number of available replicas")
+		err = extencharts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.CISBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cisbenchmark chart DaemonSets to have expected number of available nodes")
+		err = extencharts.WatchAndWaitDaemonSets(client, c.project.ClusterID, charts.CISBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+	}
+
+	cisbenchmarkChartPreUpgrade, err := extencharts.GetChartStatus(client, c.project.ClusterID, charts.CISBenchmarkNamespace, charts.CISBenchmarkName)
+	require.NoError(c.T(), err)
+
+	// Validate current version of rancher-cis-benchmark is one of the versions before latest
+	chartVersionPreUpgrade := cisbenchmarkChartPreUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Contains(c.T(), versionsList[1:], chartVersionPreUpgrade)
+
+	c.chartInstallOptions.Version, err = client.Catalog.GetLatestChartVersion(charts.CISBenchmarkName, catalog.RancherChartRepo)
```

Comment:
isn't version[0] the latest? why do we want to get the latest version again? and can we move this part before upgrade part. its a bit confusing to have it here as the next line indicates running a scan.